src/main/java/seedu/address/logic/commands/MajorSetCommand.java

```diff
@@ -0,0 +1,32 @@
+package seedu.address.logic.commands;
+
+import javafx.collections.ObservableList;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.graduation.GraduationRequirement;
+import seedu.address.model.student.Major;
+import seedu.address.model.student.Student;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class MajorSetCommand extends MajorCommand {
+    public static final String COMMAND_WORD = "set";
+    public static final String MESSAGE_NOT_IMPLEMENTED_YET = "'major set' command not implemented yet";
+    public static final String MESSAGE_USAGE = "major " + COMMAND_WORD + ": Set's the current Student's Major";
+    public static final String MESSAGE_SUCCESS = "Successfully updated Student's Major to: %1$s";
+
+    public Major major;
+
+    public MajorSetCommand(String major) {
+        this.major = new Major(major);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        model.getActiveStudent().setMajor(major);
```

Comment:
This does not seem to check if there is an active student selected before setting the major, which may lead to a similar uncaught exception. Currently, at least one student needs to exist for the planner to be valid and `planner#getActiveStudent()` tries to set a valid active student if it is `null`. However, this behavior should not be relied on because this may change in the future.
Potential bug: Uncaught exception if no student is active.